scripts/baseVehicle.gd

```diff
@@ -0,0 +1,209 @@
+extends RigidBody
+
+var vehicle_name = "unknown"
+
+
+var interface = PacketPeerUDP.new()  # UDP socket for fdm in (server)
+var peer = null
+var start_time = OS.get_ticks_msec()
+
+var last_velocity = Vector3(0, 0, 0)
+var calculated_acceleration = Vector3(0, 0, 0)
+
+
+var _initial_position = 0
+var phys_time = 0
+onready var wait_SITL = Globals.wait_SITL
+
+
+func connect_fmd_in():
+	if interface.listen(9002) != OK:
```

Comment:
port should not be hard coded, also make sure to avoid magic numbers, maybe: sitl_sim_json_port